azure/compute/src/main/java/org/jclouds/azurecompute/compute/config/ConfigurationSupplier.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c)Microsoft Open Technologies, Inc.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.compute.config;
+
+import com.google.common.base.Supplier;
+import com.microsoft.windowsazure.Configuration;
+import com.microsoft.windowsazure.management.configuration.PublishSettingsLoader;
+import org.jclouds.domain.Credentials;
+import org.jclouds.location.Provider;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.IOException;
+
+@Singleton
+public class ConfigurationSupplier implements Supplier<Configuration> {
+    private final Configuration configuration;
+
+    @Inject
+    public ConfigurationSupplier(@Provider Supplier<Credentials> creds) {
+        Configuration configuration ;
+        try {
+            configuration = PublishSettingsLoader.createManagementConfiguration(creds.get().credential, creds.get().identity);
```

Comment:
Suppliers are used to defer the work until it is actually needed, so this logic should be moved to the `get` method. If you need to cache the results of the call to get, consider decorating the supplier with `Suppliers.memoize` or similar when configuring it in the corresponding module.